src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -707,6 +779,9 @@ private static Expression BindParameterFromRouteValueOrQueryString(ParameterInfo
             return BindParameterFromValue(parameter, Expression.Coalesce(routeValue, queryValue), factoryContext);
         }
 
+        private static Expression BindParameterFromFormFile(ParameterInfo parameter, MemberExpression property, string key, FactoryContext factoryContext) =>
+            BindParameterFromValue(parameter, GetValueFromProperty(property, key, typeof(IFormFile)), factoryContext);
```

Comment:
Should this method and `BindParameterFromBody` throw if a parameter for the other is specified?

https://github.com/dotnet/aspnetcore/blob/4c8052fcc1f5ccc1c315721926ed96b553a3f7a6/src/Http/Http.Extensions/src/RequestDelegateFactory.cs#L785-L790

i.e. an action with the following signature would be invalid because it can't be both:

```csharp
app.MapPost("/blah", async ([FromBody] MyObject myJson, IFormFile myFile) =>
{
    // Whatever
});
```

Reply:
Yes it should. This wouldn't make any sense unless we natively supported binding multipart form content (we don't even support this in MVC).

Reply:
8aa7a7dc84f4f715fc05eeabe88e8e9ee553dfb3